tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/ParserTestRunner.java

```diff
@@ -0,0 +1,110 @@
+        hashSet.add("singleton_type_source_08.bal");
+        hashSet.add("singleton_type_source_01.bal");
+        hashSet.add("singleton_type_source_02.bal");
+        hashSet.add("xml_template_source_24.bal");
+        hashSet.add("explicit-new-with-object-keyword-with-multiple-args-negative02.bal");
+        hashSet.add("explicit-new-with-object-keyword-with-one-arg-negative01.bal");
+        hashSet.add("explicit-new-with-object-keyword-with-no-args.bal");
+        hashSet.add("explicit-new-with-object-keyword-with-multiple-args.bal");
+        hashSet.add("explicit-new-with-object-keyword-with-one-arg-negative02.bal");
+        hashSet.add("explicit-new-with-object-keyword-with-multiple-args-negative01.bal");
+        hashSet.add("explicit-new-with-object-keyword-with-one-args.bal");
+        hashSet.add("query_action_source_06.bal");
+        hashSet.add("send_action_source_01.bal");
+        hashSet.add("send_action_source_03.bal");
+        hashSet.add("func_def_source_08.bal");
+        hashSet.add("find_node_test_1.bal");
+        hashSet.add("enum_decl_source_08.bal");
+        hashSet.add("enum_decl_source_05.bal");
+        hashSet.add("enum_decl_source_09.bal");
+        hashSet.add("minutiae_test_05_with_no_newlines.bal");
+        hashSet.add("import_decl_source_13.bal");
+        hashSet.add("ambiguity_source_06.bal");
+        hashSet.add("ambiguity_source_11.bal");
+        hashSet.add("ambiguity_source_29.bal");
+        hashSet.add("worker_decl_source_02.bal");
+        hashSet.add("typed_binding_patterns_source_16.bal");
+        hashSet.add("forEach_stmt_source_07.bal");
+        hashSet.add("do_stmt_source_07.bal");
+        hashSet.add("match_stmt_source_13.bal");
+        hashSet.add("match_stmt_source_07.bal");
+        hashSet.add("match_stmt_source_10.bal");
+        hashSet.add("match_stmt_source_08.bal");
+        hashSet.add("match_stmt_source_12.bal");
+        hashSet.add("match_stmt_source_03.bal");
+        hashSet.add("match_stmt_source_06.bal");
+        hashSet.add("match_stmt_source_09.bal");
+        hashSet.add("match_stmt_source_11.bal");
+        return hashSet;
+    }
+
+    @DataProvider(name = "parser-test-file-provider")
+    public Object[][] dataProvider() {
+        HashSet<String> skippedTests = skipList();
+        try {
+            return Files.walk(parserDir.resolve("src").resolve("test").resolve("resources"))
+                    .filter(path -> {
+                        File file = path.toFile();
+                        return file.isFile() && file.getName().endsWith(".bal")
+                                && !skippedTests.contains(file.getName());
+                    })
+                    .map(path -> new Object[]{path.toFile().getName(), path.toString()})
+                    .toArray(size -> new Object[size][2]);
+        } catch (IOException e) {
+            Assert.fail("Can't resolve parser tests", e);
+            return new Object[0][];
```

Comment:
Can remove the try-catch block and declare the method throws IOExcepion. 
Letting it crash like that, is not a good practice right? Rather catch and Assert fail?

Reply:
Catch and Assert fail is fine.